main.cpp

```diff
@@ -24,46 +26,54 @@
 static llvm::LLVMContext context;
 
 int main() {
-	std::cout << "Hello world" << std::endl;
-	llvm::SMDiagnostic err;
-	auto file =  llvm::StringRef("/home/alex4o/bitcode_experiments_cxx/file.bc");
-	std::cout << file.str() << std::endl;
-	std::unique_ptr<llvm::Module> m = llvm::parseIRFile(file, err, context);
-	llvm::Module result(llvm::StringRef("output"), context);
-	
-	if(m)
-	{
-		m->print(llvm::outs(), nullptr);
-		for(auto &function: m->functions()) {
-			std::cout << function.getName().str() << std::endl;
-			if(function.getName().size() < 2) {
-	
-				auto resultCallee = result.getOrInsertFunction(function.getName(), function.getReturnType());
-				llvm::Function* resultFunction = llvm::dyn_cast_or_null<llvm::Function>(resultCallee.getCallee());
+  llvm::SMDiagnostic err;
+  auto file = llvm::StringRef("/home/alex4o/bitcode_experiments_cxx/file.bc");
+  std::cout << file.str() << std::endl;
+  std::unique_ptr<llvm::Module> um = llvm::parseIRFile(file, err, context);
+  const llvm::Module *m = um.get();
+
+  llvm::Module result(llvm::StringRef("output"), context);
+
+  result.setDataLayout(m->getDataLayout());
+  result.setTargetTriple(m->getTargetTriple());
+  result.setModuleInlineAsm(m->getModuleInlineAsm());
 
-				llvm::ValueToValueMapTy VMap;
-				auto resultArgs =  resultFunction->arg_begin();
-				for(llvm::Value &arg : function.args()) {
-					VMap[&*resultArgs++] = &arg;;;
-				}
-				
-				llvm::SmallVector<llvm::ReturnInst*, 8> returns;
+  llvm::ValueToValueMapTy VMap;
```

Comment:
This may by lack of understanding of C++ standards, but `VMap` thus capitalised was a slightly confusing variable name as I thought it was a type below!

Reply:
This is how the code was in the original CloneModule source code. But I understand what you mean, I would probably make all the variables start with a lowercase.

Reply:
Ah, have we copy and pasted this from somewhere?

Reply:
I am more or less trying to replicate this function: https://llvm.org/doxygen/CloneFunction_8cpp_source.html but doing it with our requirements in mind.